### ISSUE TITLE
xlC failing to compile ` typed_error<FooErrors::eFOO> foo_err`

### ISSUE THREAD

**Issue by patrickmmartin** [patrickmmartin/errorcodeNX#15]
https://github.com/patrickmmartin/errorcodeNX/blob/master/test_typed_error.cpp#L88

```typedef typed_error\<FooErrors::eFOO\> foo_err;```

and later we have 
```
  const char *msg = "foo is not a bar - thirst ensues";
  foo_err err(msg);
  foo_err err_blank; // <-- fails to compile with
```

"error_id.hpp", line 47.34: 1540-0130 (S) "FooErrors" is not declared.
"test_typed_error.cpp", line 88.11: 1540-1205 (I) The error occurred while converting to parameter 1 of "typed_error<&FooErrors::eFOO>::typed_error(const char *)".
gmake: *** [test_typed_error.ibm.o] Error 1
gmake: Target `build' not remade because of errors.

**Comment by patrickmmartin** [comment]
on the basis of

typed_error(const char* what = errtype): std::runtime_error(what) {}

https://github.com/patrickmmartin/errorcodeNX/blob/master/error_id.hpp#L47

I would expect the default of errtype to kick in here

**Comment by patrickmmartin** [comment]
looks more issues with the unspecified array?

class FooErrors {
public:
  static const char eFOO[];
};

is not good, whereas 

namespace FooErrors {
 extern error_id eFOO;
...
};

works

**Comment by patrickmmartin** [comment]
addressed in c6b0fdb8c18ec48aed2f050356f05886aafc2be2
